WCFDataService/Client/System/Data/Services/Client/DataServiceRequest.cs

```diff
@@ -272,6 +273,10 @@ internal long GetQuerySetCount(DataServiceContext context)
 
                 throw;
             }
+            finally
+            {
```

Comment:
Sorry, maybe I'm on the wrong code line. Anyway, I get your point, we need explicitly dispose in GetQuerySetCount since we do not have enumeration for $count.